torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1939,6 +1971,28 @@ def _make_tensor_helper(shape, low=None, high=None):
            sample_inputs_func=sample_inputs_slogdet,
            output_func=itemgetter(1),
            decorators=[skipCUDAIfNoMagma, skipCUDAIfRocm, skipCPUIfNoLapack]),
+    OpInfo('linalg.vector_norm',
+           op=torch.linalg.vector_norm,
+           dtypes=floating_and_complex_types_and(torch.float16, torch.bfloat16),
+           test_inplace_grad=False,
+           decorators=[skipCUDAIfNoMagma, skipCPUIfNoLapack],
+           sample_inputs_func=sample_inputs_linalg_vector_norm,
+           aten_name='linalg_vector_norm',
+           skips=(
+               # TODO: remove this once `pow` is implemented for float16
+               #       and bfloat16 on CPU. Issue:
+               #       https://github.com/pytorch/pytorch/issues/50789
+               SkipInfo('TestCommon', 'test_variant_consistency_jit',
+                        device_type='cpu',
+                        dtypes=[torch.float16, torch.bfloat16]),
+               # TODO: remove this when test_variant_consistency_jit_cuda
+               #       failures are fixed for half and bfloat16
```

Comment:
I discovered another issue here when I enabled 2-D inputs in `sample_inputs_linalg_vector_norm`. For CUDA half precision types, 2-D inputs seem to be causing the double gradient to be NaN. At least I think that's what's happening, not entirely sure yet

Reply:
The issue seems to come from here: https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_jit.py#L100

Within one of the calls to this function during `test_variant_consistency_jit_linalg_vector_norm_cuda_float16`, the magnitudes of the numbers in this multiplication are too large for float16, so the result becomes infinite:

```
l1_test:
2480.0
allSum(grads_test):
-187.875
allSum(grads_test) * l1_test:
-inf
```

One way to get around this issue is to just make the input size smaller, so I'll do that unless there's a better solution

Reply:
I had to decrease the 2-D input size for torch.half to `(2, 2)` in order to get the test to pass

Reply:
That sounds reasonable. I also have a PR that will limit this test to only run in float32. The signal we get from other dtypes isn't very good and these workaround are a pain. It hasn't landed yet, however.